### PR TITLE
bench: Benchmark MempoolToJSON

### PR DESCRIPTION
**Pull request by maflcko**
This is used in production (e.g. https://jochen-hoenicke.de/queue/#0,24h), so add a benchmark to avoid making it even slower.

Related:

* "getrawmempool true RPC call is O(n^2)" #14765

### CODE REVIEW

**Comment by promag** [comment]
Nice, concept ACK.

Changes to `entryToJSON` should be in 1st commit?

**Comment by MarcoFalke** [comment]
Thx. Force pushed with changes correctly split up into the commits

File: src/Makefile.bench.include [hunk 1]
```diff
@@ -37,6 +38,7 @@ nodist_bench_bench_bitcoin_SOURCES = $(GENERATED_BENCH_FILES)
 bench_bench_bitcoin_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES) $(EVENT_CLFAGS) $(EVENT_PTHREADS_CFLAGS) -I$(builddir)/bench/
 bench_bench_bitcoin_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 bench_bench_bitcoin_LDADD = \
+  $(LIBBITCOIN_SERVER) \
```
Comment (promag) [inline src/Makefile.bench.include]: IIUC this is the first bench touching RPC functionality.

**Comment by practicalswift** [comment]
Concept ACK

**Comment by MarcoFalke** [comment]
Anyone an idea how to fix the appveyor linker error?

cc @ken2812221

**Comment by jgarzik** [comment]
concept ACK

Comments trigged by this issue, but not directly related:

1) Recently generated univalue profiling data showed 80% of the CPU time being spent in the destructor, when testing on a large JSON dataset (https://github.com/zemirco/sf-city-lots-json)

I would infer similar behavior in getrawmempool return values.

2) Would be nice to introduce RPCv2 using protobufs a la lnd.  Clearly defined schema, automatic schema validation vs data, and dense data encoding.

**Comment by ken2812221** [comment]
Seems that bench_bitcoin does not have univalue include path. I created PR #15503 to solve it.

**Review by ryanofsky — APPROVED** [review]
utACK fa38535130266896c430d146b05d0069e525dbe6. Nice cleanup and new benchmark.

**Pull request merged**